packages/nextjs/components/scaffold-eth/Input/IntInput.tsx

```diff
@@ -0,0 +1,51 @@
+import { BigNumber, ethers } from "ethers";
+import { useCallback, useEffect, useState } from "react";
+import { InputBase } from "./InputBase";
+import { CommonInputProps, IntVariant, isValid, SIGNED_NUMBER_REGEX } from "./utils";
+
+type IntInputProps = CommonInputProps<string | BigNumber> & {
+  variant?: IntVariant;
+};
+
+export const IntInput = ({ value, onChange, name, placeholder, variant = IntVariant.INT256 }: IntInputProps) => {
+  const [inputError, setInputError] = useState(false);
+  const convertEtherToUint = useCallback(() => {
+    if (!value) {
+      return;
+    }
+    onChange(ethers.utils.parseEther(value.toString()));
+  }, [onChange, value]);
+
+  useEffect(() => {
+    if (
+      (value && typeof value === "string" && !SIGNED_NUMBER_REGEX.test(value) && value !== "-") ||
```

Comment:
Is it real to combine `IntInput` and `UintInput`? It seems the only difference is this `if` condition and a default value of `variant` prop

Reply:
Done